apps/src/feedback.js

```diff
@@ -433,8 +434,11 @@ FeedbackUtils.prototype.calculateStageProgress = function (
     newHintUsageLevels = 0.5;
   }
 
-  const newPassedProgress = newPassedLevels * 0.3 / levels.length;
-  const newPerfectProgress = newPerfectLevels * 0.4 / levels.length;
+  const passedWeight = finiteIdealBlocks ? 0.3 : 0.7;
+  const perfectWeight = finiteIdealBlocks ? 0.4 : 0;
+
+  const newPassedProgress = newPassedLevels * passedWeight / levels.length;
+  const newPerfectProgress = newPerfectLevels * perfectWeight / levels.length;
```

Comment:
as an aside, this system seems pretty inflexible; might want to drop a comment in here with a TODO to come up with something more modular